lib/azure/armrest/storage_account_manager.rb

```diff
@@ -15,9 +15,54 @@ class StorageAccountManager < ArmRestManager
       # methods for a SAM instance will return a StorageAccount object.
       def initialize(options = {})
         super
+      end
+
+      # Return information for the given storage account name for the
+      # provided +group+. If no group is specified, it will use the
+      # group set in the constructor.
+      #
+      # Example:
+      #
+      #   sam.get('portalvhdstjn1ty0dlc2dg')
+      #   sam.get('portalvhdstjn1ty0dlc2dg', 'Default-Storage-CentralUS')
+      #
+      def get(account_name, group = @resource_group)
+        set_default_subscription
+
+        raise ArgumentError, "must specify resource group" unless group
+
+        @api_version = '2014-06-01'
+        url = build_url(@subscription_id, group, account_name)
+
+        JSON.parse(rest_get(url))
+      end
+
+      # Returns a list of available storage accounts for the given subscription
+      # for the provided +group+, or all resource groups if none is provided.
+      #
+      def list(group = @resource_group)
+        if group
+          @api_version = '2014-06-01'
+          url = build_url(@subscription_id, group)
+          JSON.parse(rest_get(url))['value'].first
+        else
+          arr = []
+          thr = []
+
+          resource_groups.each do |group|
+            @api_version = '2014-06-01' # Must be set after resource_groups call
+            url = build_url(@subscription_id, group['name'])
+
+            thr << Thread.new{
+              res = JSON.parse(rest_get(url))['value'].first
+              arr << res if res
+            }
+          end
 
-        @base_url += "resourceGroups/#{@resource_group}/"
-        @base_url += "providers/Microsoft.Storage/storageAccounts"
+          thr.each{ |t| t.join }
```

Comment:
Prefer `thr.each(&:join)`.  Also why the abbreviations?  I think a variable called `threads` would be much clearer.